website/case-studies/implementation-of-flint-modules.md

```diff
@@ -0,0 +1,407 @@
+
+## Background
+
+It is useful to think about what a module manifest might look like. **Manifests** are meta-data used by the package managers to describe the compatibility (FLINT versions that are used), dependencies (necessary inputs) and the configuration requirements (for example whether the module uses daily, monthly or annual time steps).
+
+This case study describes the core contribution of each module.
+
+## Index
+
+1.  [Prerequisites](#prerequisites)
+2.  [How do modules work?](#how-do-modules-work)
+3.  [Empirical forest growth module](#empirical-forest-growth-module)
+4.  [Hybrid forest growth module](#hybrid-forest-growth-module)
+5.  [WOFOST crop growth module](#wofost-crop-growth-module)
+6.  [Turnover module](#turnover-module)
+7.  [Decomposition module](#decomposition-module)
+8.  [RothC soil carbon module](#rothc-soil-carbon-module)
+9.  [Chapman Richards module](#chapman-richards-module)
+10. [GCBM module](#gcbm-module)
+11. [Frequently asked questions](#frequently-asked-questions)
+
+## Prerequisites
+
+These are the prerequistes you need to know before learning about **Modules**. This section presumes that you are already familiar with **FLINT**. If not, see [the FLINT repository](https://github.com/moja-global/FLINT) to better understand **FLINT** before continuing.
+
+## How do modules work?
+
+Each module reads (or is provided with – see Unit Controller) information about the current state variables and the data required to update the state variables, such as climate data or information about disturbances to simulate. Each module then performs the required calculations and returns information about the updates to apply to each of the state variables and C pools. 
+
+For state variables such as age or height it is possible for modules to return the updated value but for all carbon pools and other fluxes the module returns an array (sparse matrix) of the proposed operations. This array will include the information about the source pool, the sink pool, and the amount (Mg ha-1 per time step) of the flux. Module-specific metadata regarding units and time step size are also required. This information is all returned to the Unit Controller.
+
+In terms of the FLINT programming framework, there are internal and external modules. The internal modules tend to be generic utility modules, for example, for producing output tables. While external modules are regionally or nationally specific. By using modules and separating the generic and specific, FLINT’s framework remains highly flexible and new modules can be ‘plugged in’ as they are developed.
```

Comment:
It might help to add a short summary of what is going to be described next - currently, there isn't a lot of continuity between the topics which can be jarring for the reader.

Reply:
> Adding or developing modules is an advanced topic, which will not be covered here. If you would like to implement a new module, please reach out on #modules-and-gbcm on Slack.